augur/utils.py

```diff
@@ -259,6 +265,114 @@ def _read_nuc_annotation_from_gff(record, reference):
+    # Matching Nextclade conventions (NAME_ATTRS_CDS)
+    # https://github.com/nextstrain/nextclade/blob/59e757fd9c9f8d8edd16cf2063d77a859d4d3b96/packages/nextclade/src/io/gff3.rs#L35-L54
+    QUALIFIER_PRIORITY = [ "Name", "name", "Alias", "alias", "standard_name", "old-name", "Gene", "gene", "gene_name",
+        "locus_tag", "product", "gene_synonym", "gb-synonym", "acronym", "gb-acronym", "protein_id", "ID"]
+
+    for qualifier in QUALIFIER_PRIORITY:
+        if qualifier in feature.qualifiers:
+            return feature.qualifiers[qualifier][0]
+    raise AugurError(f"No valid feature name found for feature for feature {feature.id}")
+
+def _read_gff_like_nextclade(reference, feature_names) -> Dict[str, SeqFeature]:
+    """
+    Read a GFF file the way Nextclade does. That means:
+    - We only look at CDS features.
+    - CDS features with identical IDs are joined into a compound feature.
+    - The feature name is taken from qualifiers in the same priority order as in Nextclade.
+
+    Parameters
+    ----------
+    reference : string
+        File path to GFF reference
+    feature_names : None or set or list
+        Restrict the genes we read to those in the set/list
+
+    Returns
+    -------
+    features : dict
+        keys: feature names, values: :py:class:`Bio.SeqFeature.SeqFeature`
+        Note that feature names may not equivalent to GenBank feature keys
+
+    Raises
+    ------
+    AugurError
+        If the reference file contains no IDs or multiple different seqids
+        If a gene is found with the name 'nuc'
+    """
+    rec = _load_gff_record(reference)
+    features = {'nuc': _read_nuc_annotation_from_gff(rec, reference)}
+
+    cds_features: Dict[str, SeqFeature] = {}
+
+    def _flatten(feature):
+        if feature.type == "CDS":
+            if not feature.id:
+                # Hack to ensure we have some IDs for all features
+                # Technically, all features should have an ID, but Nextclade doesn't require it
+                feature.id = str(feature.qualifiers)
+            if feature.id in cds_features:
+                if isinstance(cds_features[feature.id].location, CompoundLocation):
+                    cds_features[feature.id].location.parts.append(feature.location)
+                else:
+                    cds_features[feature.id].location = CompoundLocation([cds_features[feature.id].location, feature.location])
+            else:
+                cds_features[feature.id] = feature
+        for sub_feature in getattr(feature, "sub_features", []):
+            _flatten(sub_feature)
+
+    for feature in rec.features:
+        _flatten(feature)
+    
+    for feature in cds_features.values():
+        feature_name = _lookup_feature_name_like_nextclade(feature)
+        if feature_name == 'nuc':
+            raise AugurError(f"Reference {reference!r} contains a gene with the name 'nuc'. This is not allowed.")
+        if feature_name in feature_names or feature_names is None:
```

Comment:
```suggestion
        if feature_names is None or feature_name in feature_names:
```

from #1668 import bugfix

Thanks @jameshadfield for the suggestion